Testing

theodorebugnet opened the pull request:
Adds unit tests for all the stats APIs. Excludes the report APIs since those are not actively used or maintained right now.

Since the APIs are all only reading from the database, the docker-compose does not need to be restarted between tests.

Commit:
WIP testing

Commit:
feat(testing): proper testing with working docker setup

Commit:
feat(testing): fixed issue/redeem tests

Commit:
Merge branch 'master' of github.com:interlay/polkabtc-stats into theo/testing

Commit:
chore(testing): cleanup unused packages/commands

Commit:
Merge branch 'master' of github.com:interlay/polkabtc-stats into theo/testing

nud3l commented on test/issue/issue.test.ts:
```diff
@@ -0,0 +1,113 @@
+        const dailyAmount = await getRecentDailyIssues(3);
+        const satoshi = "3093882237";
+        const nextMidnight = getUTCMidnight(new Date()).getTime();
+        const expected = Array.from({length: 4}, (_, idx) => ({date: nextMidnight - (idx * 86400 * 1000), sat: satoshi} as SatoshisTimeData)).reverse();
```
minor nit: could break the line for better readability

nud3l reviewed (commented):
Overall, this is already very helpful. One question I have is why did you decide to stringify some of the number types? If the values are too large, we should use `BN` for integers and for any float, fixed points etc. `Big`.
Eventually, I would like to have the type definitions in the lib so that we can just make changes there when we do breaking changes on the parachain (minus the event parsing).

theodorebugnet commented:
I found that some values were actually always being returned from the database as strings rather than numbers (so the model types were innacurate). You make a good point that they could be parsed immediately, returning a number from the API and with error handling right there in case of an invalid string.
Should I refactor to number/BN/Big right now, or put it off until we have more centralised types in the lib?

nud3l commented:
I think refactoring now is likely best. In most places, we should already have the BN/Big types in the lib and the UI so that would simplify harmonizing the code bases.

Commit:
fix(tests): refactored certainl models from string to number.
JS raw numbers appear safe for satoshi denominations up till approx. 90M BTC, hence BN wasn't used.

nud3l commented on src/redeem/redeemDataService.ts:
```diff
@@ -125,7 +125,7 @@ export async function getRecentDailyRedeems(
                 [daysBack]
             )
-        ).rows.map((row) => ({ date: row.date, sat: row.sat }));
+        ).rows.map((row) => ({ ...row, sat: Number(row.sat) }));
```
Satoshis might not fit in `number`. Better to use `BN`

theodorebugnet replied:
I checked and `100_000_000 * 21_000_000 < Number.MAX_SAFE_INTEGER` so it should be good.

nud3l commented on src/vaults/vaultModels.ts:
```diff
@@ -2,12 +2,12 @@ import { DistributionStats } from "../common/commonModels";
 export interface CollateralTimeData {
     date: number;
-    amount: number;
+    amount: string;
```
Do you want to have the amount as a string here rather than a a `Big` if it's denoted in DOT or `BN` if it's denoted in planck?

theodorebugnet replied:
Dangit I thought I went through to check any other changes where I used a string, but missed this file. Will do.

Commit:
fix(tests): collateral to BN

Commit:
Merge branch 'master' of github.com:interlay/polkabtc-stats into theo/testing
